spec/forms/concerns/form_attributes_spec.rb

```diff
@@ -0,0 +1,129 @@
+require 'rails_helper'
+
+class ExampleModel
+  # This was copied from OutgoingEmail to have a real example, but not literally
+  # tie to an existing model (which may change)
+  def self.column_defaults
+    {
+      "id" => nil,
+      "body" => "some body text",
+      "client_id" => nil,
+      "created_at" => nil,
+      "mailgun_status" => "sending",
+      "message_id" => nil,
+      "sent_at" => nil,
+      "subject" => nil,
+      "to" => nil,
+      "updated_at" => nil,
+      "user_id" => nil
+    }
+  end
+end
+
+class SecondExampleModel
+  # This was copied from OutgoingEmail to have a real example, but not literally
+  # tie to an existing model (which may change)
+  def self.column_defaults
+    {
+      "id" => nil,
+      "body" => "some body text",
+      "client_id" => nil,
+      "created_at" => nil,
+      "mailgun_status" => "sending",
+      "message_id" => nil,
+      "sent_at" => nil,
+      "subject" => nil,
+      "to" => nil,
+      "updated_at" => nil,
+      "user_id" => nil
+    }
+  end
+end
+
+class SymbolExampleForm
+  include ActiveModel::Model
+  include ActiveModel::Validations::Callbacks
+  include FormAttributes
+
+  set_attributes_for :empty
+  set_attributes_for :without_defaults, :attribute_one, :attribute_two
+  set_attributes_for :with_defaults,
+    :default_attribute_one,
+    :default_attribute_two,
+    :default_attribute_three,
+    defaults: {
+      default_attribute_one: 'foo',
+      default_attribute_two: 'bar',
+    }
+end
+
+class ModelExampleForm
+  include ActiveModel::Model
+  include ActiveModel::Validations::Callbacks
+  include FormAttributes
+
+  set_attributes_for ExampleModel,
+    :body,
+    :client_id,
+    :mailgun_status,
+    defaults: {
+      mailgun_status: "not_sent",
+    }
+
+  set_attributes_for SecondExampleModel,
+    :body
+end
+
+
+RSpec.describe FormAttributes do
+  describe "#set_attributes_for" do
+    context "when passing a symbol" do
+      subject { SymbolExampleForm.new }
+
+      it 'should create an empty hash by default' do
+        attributes = subject.attributes_for(:empty)
+        expect(attributes).to be_a Hash
+        expect(attributes).to be_empty
+      end
+
+      it 'should create initialize with nil if defaults are not passed' do
+        attributes = subject.attributes_for(:without_defaults)
+        expect(attributes.keys).to eq [:attribute_one, :attribute_two]
+        expect(attributes.values).to be_all nil
```

Comment:
i think this also works! i think what's nice about it is seeing it as a hash but i don't really have a preference 
```suggestion
        expect(attributes).to match({attribute_one: nil, attribute_two: nil})
```

Reply:
(same goes for the following test blocks, ofc)

Reply:
i agree this would be clearer